test/common/testlib.py

```diff
@@ -1330,11 +1330,6 @@ def filterOutput(self, test, failed, output):
         tries += 1
         setattr(test, "retryCount", tries)
 
-        # Didn't fail or retried too much, just print output and continue
-        if tries >= 3 or not failed:
```

Comment:
If it didn't fail there is not need to run tests-policy but with this change we will run it anyway.
Or am I getting something wrong?

Reply:
Ah right, I missed that.
I updated the code, so it just returns when it did not fail and does retrying only when it has less than 3 retries.